File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPVelocity.java
package org.noxet.noxetserver.playerstate.properties;

import org.bukkit.entity.Player;
import org.bukkit.util.Vector;
import org.noxet.noxetserver.playerstate.PlayerStateProperty;

public class PSPVelocity implements PlayerStateProperty<Vector> {
    @Override
    public String getConfigName() {
        return "velocity";
    }

    @Override
    public Vector getDefaultSerializedProperty() {
        return new Vector();
    }

    @Override
    public Vector getSerializedPropertyFromPlayer(Player player) {
        return player.getVelocity();
    }

    @Override
    public void restoreProperty(Player player, Vector velocity) {
        player.setVelocity(velocity);
    }
}

File: src/main/java/org/noxet/noxetserver/playerstate/properties/PSPVelocity.scala
package org.noxet.noxetserver.playerstate.properties

import org.bukkit.entity.Player
import org.bukkit.util.Vector
import org.noxet.noxetserver.playerstate.PlayerStateProperty

object PSPVelocity extends PlayerStateProperty[Vector]:
  override def getConfigName: String = "velocity"

  override def getDefaultSerializedProperty: Vector = Vector()

  override def getSerializedPropertyFromPlayer(player: Player): Vector =
    player.getVelocity

  override def restoreProperty(player: Player, velocity: Vector): Unit =
    player.setVelocity(velocity)